python/ray/data/tests/test_dataset.py

```diff
@@ -3973,6 +3973,47 @@ def get_node_id():
     assert set(locations) == {node1_id, node2_id}
 
 
+def test_parquet_read_spread_no_cpus(ray_start_cluster, tmp_path):
+    cluster = ray_start_cluster
+    cluster.add_node(
+        resources={"foo": 100}, _system_config={"max_direct_call_object_size": 0}
+    )
+    cluster.add_node(resources={"bar:1": 100})
+    cluster.add_node(resources={"bar:2": 100}, num_cpus=0)
```

Comment:
maybe add GPU as well?

Reply:
Doesn't matter for this test. This test is ensuring that a task with the spread resource prefix and with `num_cpus=0` is able to be placed on a node with 0 CPUs advertised, so it's orthogonal to whether or not it has any GPUs.